cmd/new_test.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cmd
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/pulumi/pulumi/pkg/backend"
+	"github.com/pulumi/pulumi/pkg/backend/display"
+	"github.com/pulumi/pulumi/pkg/workspace"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreatingProjectWithSpecifiedName(t *testing.T) {
+	tempdir, _ := ioutil.TempDir("", "test-env")
+	defer os.RemoveAll(tempdir)
+	assert.NoError(t, os.Chdir(tempdir))
+	uniqueProjectName := filepath.Base(tempdir)
+
+	var args = newArgs{
+		interactive:       false,
+		name:              uniqueProjectName,
+		prompt:            promptForValue,
+		secretsProvider:   "default",
+		templateNameOrURL: "typescript",
+	}
+
+	err := runNew(args)
+	assert.NoError(t, err)
+
+	removeStack(t)
+
+	proj := loadProject(t, tempdir)
+	assert.Equal(t, uniqueProjectName, proj.Name.String())
+}
+
+func TestCreatingProjectWithEnteredName(t *testing.T) {
```

Comment:
What is the difference between `SpecifiedName` and `EnteredName`? In both of these tests `uniqueProjectName` is being specified as a the `name` arg.

Reply:
Great catch! I wasn't careful enough and let it slip in. The `Entered` ones shouldn't have `name` specified.

Reply:
Just a nit, but can we rename the functions to make the distinction more clear? Maybe something like:

 - TestCreatingProjectWith**Args**SpecifiedName
 - TestCreatingProjectWith**Prompted**Name

Reply:
Done